### DISCUSSION TITLE
SQS Error handling

### DISCUSSION THREAD

**Discussion by fabian-grajko** [aws/aws-sdk-js-v3#3563] [General]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js/issues) and didn't find any solution
  
**Describe the bug**
SQSClient throws `TypeError` when Command input is incorrect

**Is the issue in the browser/Node.js?**
Node.js

**If on Node.js, are you running this on AWS Lambda?**
No

**Details of the browser/Node.js version**
v14.16.1

**SDK version number**
"@aws-sdk/client-sqs": "3.18.0"

**To Reproduce (observed behavior)**
- Initialize a new instance of SQSClient
- Create Command with an incorrect input
- `.send()` invalid command
```
(node:9571) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'Code' of undefined
    at loadQueryErrorCode (./node_modules/@aws-sdk/client-sqs/dist/cjs/protocols/Aws_query.js:2583:20)
    at deserializeAws_querySendMessageCommandError (./node_modules/@aws-sdk/client-sqs/dist/cjs/protocols/Aws_query.js:1024:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Expected behavior**
A valid error is thrown.

**Additional context**
I tested it with `SendMessageBatchCommand` and `SendMessageCommand`

**Comment by ajredniwja** [comment]
@fabian-grajko thanks for opening this, transferred it to the correct repo. Can you please share the specific code that you are using?

**Comment by fabian-grajko** [comment]
Sure. I reproduced the error using `localstack`.
```javascript
const { SQSClient, SendMessageCommand } = require('@aws-sdk/client-sqs');

const client = new SQSClient({
  endpoint: 'http://localhost:4566',
  region: 'us-east-2',
});

const command = new SendMessageCommand({
  QueueUrl: 'http://localhost:4566/000000000000/test.fifo',
  MessageBody: undefined, // This line is causing problem
  MessageGroupId: '1',
});

client.send(command).then(console.log).catch(console.error);
```

Output:
```
TypeError: Cannot read property 'Code' of undefined
    at loadQueryErrorCode (./node_modules/@aws-sdk/client-sqs/dist/cjs/protocols/Aws_query.js:2583:20)
    at deserializeAws_querySendMessageCommandError (./node_modules/@aws-sdk/client-sqs/dist/cjs/protocols/Aws_query.js:1024:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Comment by ajredniwja** [comment]
@fabian-grajko I get it now, just in general not a good idea to pass `undefined` there, with that being said, what error do you think should be thrown?

**Comment by fabian-grajko** [comment]
If `MessageGroupId` is invalid, the client throws `MissingParameter: The request must contain the parameter MessageGroupId.`
It would be nice to get a similar error for `MessageBody`

**Comment by dormeiri** [comment]
I also have this issue when sending `DeleteMessageBatchCommand` with a non-existing receipt handle:
```javascript
    const request: DeleteMessageBatchRequest = {
        QueueUrl: queueUrl,
        Entries: [{Id: 'msg1', ReceiptHandle: 'not-found'}]
    };
    const command = new DeleteMessageBatchCommand(request);
    await this.client.send(command); // Throws "Cannot read property 'Code' of undefined"
```

It also occurs in `DeleteMessageCommand`
